src/main/java/org/jabref/logic/preferences/JabRefCliPreferences.java

```diff
@@ -960,6 +961,32 @@ public void exportPreferences(Path path) throws JabRefException {
     public void importPreferences(Path file) throws JabRefException {
         try (InputStream is = Files.newInputStream(file)) {
             Preferences.importPreferences(is);
+            citationKeyPatternPreferences = null;
```

Comment:
~~Please add a comment above stating that `null`in will trigger re-reading the preferences.~~

Extract method `forcePreferenceCacheRefresh` and put all setting `null` to there.